### DISCUSSION TITLE
Pre-splitting a 3D heat structure

### DISCUSSION THREAD

**Discussion by TheBEllis** [idaholab/moose#28600] [Q&A Modules: Thermal Hydraulics]
Hi MOOSE team,
I am looking to do a thermal hydraulics calculation using a fairly large 3D heat structure. I was wondering if there is functionality in MOOSE to pre-split the heat structure mesh? I imagine the whole mesh is needed for heat transfer between the 1D components, making distributed runs fairly difficult. But I just thought I would ask. Thanks!

**Comment by GiudGiud** [comment]
@joshuahansel

**Comment by joshuahansel** [comment]
I am not sure - I don't know if anyone has tried this in THM. I will try to make a test for this setup soon, just to see IF it works. I'm not sure about how much effort would be required to fix it if it does not work already. What's your timeline?

**Reply by TheBEllis** [reply]
I don't have a timeline for this at the moment, it was more of a test to see what I could get to work, so no rush. Let me know if you give it a go:)
